txp/Pos/Txp/Toil/Utxo/Functions.hs

```diff
@@ -115,6 +118,21 @@ verifyTxUtxo ctx@VTxContext {..} ta@(TxAux UnsafeTx {..} witnesses) = do
         verifyConsistency _txInputs witnesses
         verifyOutputs ctx ta
 
+    filterBlacklisted
+        :: NonEmpty (TxIn, TxOutAux)
+        -> ExceptT ToilVerFailure UtxoM (NonEmpty (TxIn, TxOutAux))
+    filterBlacklisted xs =
+        case NE.filter notBlacklistedSrcAddr xs of
+            [] -> throwError ToilEmptyAfterFilter
+            ys -> pure $ NE.fromList ys
```

Comment:
can we use the safe constructor and pattern match on `ys`? I know that this is safe because we're pattern matching on `[]` above, but my spidey sense tingles whenever I see `NE.fromList` :\ `(y:ys) -> pure (y :| ys)`